### DISCUSSION TITLE
I would like to inquire whether the IBM HMC and VIO are compatible with the GLPI agent.

### DISCUSSION THREAD

**Discussion by psl2024** [glpi-project/glpi-agent#682] [Q&A] [answered]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

I am attempting to perform a remote inventory for IBM HMC and VIO, but I always encounter an error:

[info] running task RemoteInventory
[debug] Remote mode enabled: ssh
[debug2] executing ssh -q -o BatchMode=yes -l hscroot -o ConnectTimeout=180 x.x.x.x LANG=C id -u
[debug] Skipping remote inventory task execution for x.x.x.x-2024-05-28-09-56-29: Can't run simple command on remote, check server is up and ssh access is setup
[debug] Remote inventory task run in 1 seconds

The command I use:
glpi-agent --remote="ssh://xxxx:"xxxxxx"@x.x.x.x?mode=ssh" --logger=stderr --tasks remoteinventory --debug --debug

However, when I try to use SSH (ssh xxx@x.x.x.x) to test, it is successful.

**Comment by g-bougard** [comment, marked as answer]
Hi @psl2024 

it seems you're trying to access your remote device using a password but only in ssh mode. This doesn't work. The ssh mode will only work with ssh public key authentication. Password based authentication requires libssh2.

That said, I don't know if IBM HMC & VIO are well supported via ssh remote inventory.

**Reply by psl2024** [reply]
Hi @g-bougard 
Thank you for your help. I think for now I will add them manually.
